--- a/template/server/main.py
+++ b/template/server/main.py
@@ -41,10 +41,12 @@
         default_websockets["python"] = python_context.id
         websockets["default"] = websockets[python_context.id]
 
-        javascript_context = await create_context(
-            client, websockets, "javascript", "/home/user"
-        )
-        default_websockets["javascript"] = javascript_context.id
+        kernelspecs = (await client.get(f"{JUPYTER_BASE_URL}/api/kernelspecs")).json()
+        if "javascript" in kernelspecs.get("kernelspecs", {}):
+            javascript_context = await create_context(
+                client, websockets, "javascript", "/home/user"
+            )
+            default_websockets["javascript"] = javascript_context.id
 
         logger.info("Connected to default runtime")
         yield
